modules/core/src/main/java/org/apache/ignite/internal/processors/cache/StateChangeRequest.java

```diff
@@ -111,7 +109,7 @@ public ClusterState prevState() {
      * @return {@code True} if active state was changed.
      */
     public boolean activeChanged() {
-        return active(prevState) && !active(msg.state()) || !active(prevState) && active(msg.state());
+        return prevState.active() ^ msg.state().active();
```

Comment:
I think `prevState.active() != msg.state().active()` is more readable. WDYT?

Reply:
Agree